src/ios/FirebaseAnalyticsPlugin.m

```diff
@@ -101,21 +101,30 @@ - (void)requestTrackingAuthorization:(CDVInvokedUrlCommand *)command {
 - (void)showTrackingAuthorizationPopup:(CDVInvokedUrlCommand *)command {
     
     if (@available(iOS 14, *)) {
-        [ATTrackingManager requestTrackingAuthorizationWithCompletionHandler:^(ATTrackingManagerAuthorizationStatus status) {
-            BOOL result = false;
-            switch(status) {
-                case ATTrackingManagerAuthorizationStatusAuthorized: {
-                    result = true;
-                    break;
-                }
-                default: {
-                    result = false;
-                    break;
+        
+        NSDictionary *dict = NSBundle.mainBundle.infoDictionary;
+        
+        if([dict objectForKey:@"NSUserTrackingUsageDescription"]){
+            [ATTrackingManager requestTrackingAuthorizationWithCompletionHandler:^(ATTrackingManagerAuthorizationStatus status) {
+                BOOL result = false;
```

Comment:
you don't need the switch, you just need to do `BOOL result = status == ATTrackingManagerAuthorizationStatusAuthorized`

Reply:
Done